pkg/controllers/cloud_config_sync_controller_test.go

```diff
@@ -224,10 +224,12 @@ var _ = Describe("Cloud config sync controller", func() {
 		co := &configv1.ClusterOperator{}
 		err := cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, co)
 		if err == nil || !apierrors.IsNotFound(err) {
-			Eventually(func() bool {
-				err := cl.Delete(context.Background(), co)
-				return err == nil || apierrors.IsNotFound(err)
-			}).Should(BeTrue())
+			Eventually(func() error {
+				return cl.Delete(context.Background(), co)
+			}).Should(SatisfyAny(
+				Not(HaveOccurred()),
+				MatchError(apierrors.IsNotFound, "IsNotFound"),
```

Comment:
Not seen this used like this before, but cool, checked the docs and it works, nice find

Reply:
Might be new. I'm a little surprised there's not a matcher already defined that does this same thing, but I wasn't able to find one.